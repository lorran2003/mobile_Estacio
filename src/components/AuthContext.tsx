import React, { createContext, PropsWithChildren, useContext, useState } from 'react'
import { SessionUser } from '../type/SessionUser';
import { DataUser } from '../type/DataUser';
import { SignIn, signInUser } from '../API/User/signInUser';
import { router } from 'expo-router';
import * as ImagePicker from 'expo-image-picker';


const AuthContextTag = createContext<SessionUser>({
    'isLoading': true,
    'user': null,
    'session': false,
    'image': null,
    signIn: () => null,
    signOut: () => null,
    pickImage: () => null
});

// This hook can be used to access the user info.
export function useSession() {
    const value = useContext(AuthContextTag);
    if (process.env.NODE_ENV !== 'production') {
        if (!value) {
            throw new Error('useSession must be wrapped in a <SessionProvider />');
        }
    }

    return value;
}

export default function AuthContext({ children }: PropsWithChildren) {

    const [user, setUser] = useState<DataUser | null>(null);

    const [isLoading, setIsLoading] = useState<boolean>(false);

    const [session, setSession] = useState<boolean>(false);

    const [image, setImage] = useState<any>(null);


    return (
        <AuthContextTag.Provider value={
            {
                'isLoading': isLoading,
                'session': session,
                'user': user,
                'image': image,
                signIn: async ({ email, password }: SignIn) => {

                    setIsLoading(true);

                    const dataUser: SignIn = {
                        'email': email,
                        'password': password
                    }

                    try {
                        const response = await signInUser(dataUser);

                        if (!response) {
                            throw new Error('Usuário não encontrado');
                        }

                        setImage(response.photo);

                        setUser(response);
                        setSession(true);
                        router.replace('/(tabs)/user/profile');

                    }
                    catch (error) {
                        console.error(error);
                    }
                    finally {

                        setIsLoading(false);
                    }
                },

                signOut: () => {
                    setSession(false);
                    setUser(null);
                    setImage(null);
                    router.push('/(tabs)/');
                },
                pickImage: async () => {
                    // No permissions request is necessary for launching the image library
                    let result = await ImagePicker.launchImageLibraryAsync({
                        mediaTypes: ImagePicker.MediaTypeOptions.All,
                        allowsEditing: true,
                        aspect: [4, 3],
                        quality: 1,
                    });

                    if (!result.canceled) {
                        setImage(result.assets[0].uri);
                    }
                }
            }

        }
        >
            {children}

        </AuthContextTag.Provider>
    )
}