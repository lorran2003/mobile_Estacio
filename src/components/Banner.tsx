import { View, Text, Image } from 'react-native'

export function Banner() {
    return (
        <View className='w-full h-56'>
            <Image
                source={require('../assets/images/banner.png')}
                className='w-full h-full'
            />
        </View>
    )
}