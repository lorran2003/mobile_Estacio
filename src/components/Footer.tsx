import { AntDesign, FontAwesome } from '@expo/vector-icons'
import { Link } from 'expo-router'
import { View, Text } from 'react-native'

export function Footer() {
    return (
        <View className='w-full h-40 justify-center items-center gap-3'>

            <View className='flex-row gap-3'>

                <Link
                    href={'/favorite'}
                    className='text-zinc-800/70'
                >
                    favoritos
                </Link>
               
                <Link
                    href={'/cart'}
                    className='text-zinc-800/70'
                >
                    Carrinho
                </Link>

                <Text className='text-zinc-800/70'>promos</Text>

                <Link
                    href={'/(user)/profile'}
                    className='text-zinc-800/70'
                >
                    minha conta
                </Link>
                
                <Text className='text-zinc-800/70'>ajuda</Text>

            </View>

            <View className='flex-row gap-5'>
                <AntDesign name="instagram" size={24} color="black" />
                <FontAwesome name="whatsapp" size={24} color="black" />
            </View>

            <Text className='text-zinc-800/70'>Â© 2024 Dazala, Inc. All rights reserved
            </Text>
        </View>
    )
}