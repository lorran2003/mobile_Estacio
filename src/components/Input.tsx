import { Text, TextInput, View } from 'react-native'

interface PropsInput {
    label: string;
    onChangeText: (text: string) => void;
    length?: number;
}

export function Input({ label, onChangeText, length }: PropsInput) {
    return (
        <View className='gap-2 justify-center items-centers'>
            <Text className='text-xl opacity-60'>{label}</Text>
            <TextInput
                cursorColor={'#000'}
                className='rounded-lg border border-[#CA9D37] w-72 h-12 p-2'
                onChangeText={(text) => onChangeText(text)}

            />
        </View>
    )
}