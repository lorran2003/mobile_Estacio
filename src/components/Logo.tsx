import { View, Image, StyleSheet } from 'react-native'

export function Logo() {
    return (
        <View className='size-24 items-center justify-center rounded-full border-solid border-[#CA9D37] border'>
            <Image
                source={require('../assets/images/logo.jpg')}
                style={style.container}
            />
        </View>
    )
}

const style =  StyleSheet.create({
    container: {
        width:"100%",
        height:"100%",
        objectFit:"cover",
        borderRadius:9999
    }
})