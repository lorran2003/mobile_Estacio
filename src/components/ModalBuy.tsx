import { View, Text, Modal, Image, Pressable, StyleSheet, ScrollView, Alert } from 'react-native'
import { PropsAPI } from '../type/typeAPI';
import { TitlePages } from './TitlePages';
import { AntDesign } from '@expo/vector-icons';
import { addToCart } from '../API/cart/addToCart';

interface PropsModalBuy {
  product: PropsAPI;
  visible: boolean;
  closeModal: () => void;
}

export function ModalBuy({ product, visible, closeModal }: PropsModalBuy) {

  const addProductToCart = async ({ product }: { product: PropsAPI }) => {
    try {

      await addToCart({ product });
      Alert.alert("Produto adicionado ao carrinho!");
      closeModal();

    } catch (error) {
      console.error("Erro ao adicionar ao carrinho : " + error);
    }
  }

  return (
    <Modal
      visible={visible}
      animationType='slide'
      onRequestClose={() => closeModal()}
      transparent={true}
    >

      <ScrollView className='h-full bg-zinc-800/70'>

        <View className='items-center justify-start h-full gap-5 pb-5'>

          <TitlePages title={product.name} />

          <View className='w-11/12 justify-center items-end'>

            <Pressable
              className='rounded-full bg-zinc-50'
              onPress={() => closeModal()}
            >
              <AntDesign name="close" size={30} color="black" />
            </Pressable>

          </View>

          <View className='w-11/12 bg-zinc-50 rounded-md gap-5 p-3'>

            <Image
              source={{ uri: product.image }}
              className='w-full h-96 rounded-md'
            />

            <View className='gap-3'>

              <Text className='font-semibold text-2xl'>R$ {product.price}</Text>

              <View className='w-full flex-row flex-wrap items-start justify-between gap-2'>

                <View className='gap-2 justify-center items-start'>

                  <Text className='text-zinc-800/70'>Selecione o tamanho</Text>

                  <View className='flex-row gap-2'>

                    <Pressable className='size-10 justify-center items-center bg-neutral-200 rounded-md'>
                      <Text>PP</Text>
                    </Pressable>

                    <Pressable className='size-10 justify-center items-center bg-neutral-200 rounded-md'>
                      <Text>P</Text>
                    </Pressable>

                    <Pressable className='size-10 justify-center items-center bg-neutral-200 rounded-md'>
                      <Text>M</Text>
                    </Pressable>

                    <Pressable className='size-10 justify-center items-center bg-neutral-200 rounded-md'>
                      <Text>G</Text>
                    </Pressable>

                  </View>

                </View>

                <View className='gap-2' >

                  <Pressable
                    className='bg-zinc-800 rounded-md p-3 flex-row gap-2 justify-center items-center'
                    onPress={() => addProductToCart({ product })}
                  >
                    <Text className='text-zinc-50'>Adicionar</Text>
                    <AntDesign name="shoppingcart" size={24} color="white" />
                  </Pressable>

                  <Pressable className='bg-[#CA9D37] rounded-md p-3 justify-center items-center'>
                    <Text className='text-zinc-50'>Comprar</Text>
                  </Pressable>

                </View>
              </View>
            </View>
          </View>
        </View>
      </ScrollView>
    </Modal>
  )
}