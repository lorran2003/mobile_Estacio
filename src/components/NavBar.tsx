import { View, Text, Pressable, Image } from 'react-native'
import { AntDesign, Ionicons } from '@expo/vector-icons'
import { Link } from 'expo-router'

export function NavBar() {
    return (
        <View className='flex-row justify-between items-center w-full p-5'>
            <Link href={'/'}>
                <View className='items-center justify-center shadow shadow-zinc-800 rounded-full border-solid border-[#CA9D37] border'>
                    <Image
                        source={require('../assets/images/logo.jpg')}
                        className='size-20 rounded-full'
                    />
                </View>
            </Link>

            <View className='flex-row gap-2'>
                <Pressable className='border-solid border-[#CA9D37] border bg-zinc-50 shadow shadow-zinc-800 p-2 rounded-full'>
                    <AntDesign name="user" size={24} color="black" />
                </Pressable>

                <Pressable className='border-solid border-[#CA9D37] border bg-zinc-50 shadow shadow-zinc-800 p-2 rounded-full'>
                    <AntDesign name="shoppingcart" size={24} color="black" />
                </Pressable>

                <Pressable className='border-solid border-[#CA9D37] border bg-zinc-50 shadow shadow-zinc-800 p-2 rounded-full'>
                    <Ionicons name="menu-outline" size={24} color="black" />
                </Pressable>
            </View>
        </View>
    )
}