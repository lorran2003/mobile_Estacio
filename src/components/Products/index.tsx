import { AntDesign } from '@expo/vector-icons'
import { Link } from 'expo-router'
import { View, Text } from 'react-native'
import { CarouselProducts } from './CarouselProducts'

export function ProductsHome() {
    return (
        <View className='justify-center items-center gap-3 pb-5 w-full'>

            <Text className='text-[##CA9D37] text-4xl font-bold'>Nossos produtos</Text>

            <Link href={'/products'}>
                <View className='flex-row justify-center items-center gap-1'>
                    <Text className='underline'>
                        Ver todos os produtos
                    </Text>
                    <AntDesign name="arrowright" size={20} color="black" />
                </View>
            </Link>

            <CarouselProducts />

        </View>
    )
}