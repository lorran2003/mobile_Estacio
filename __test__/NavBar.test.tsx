import { NavBar } from '@/src/components/NavBar';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
});