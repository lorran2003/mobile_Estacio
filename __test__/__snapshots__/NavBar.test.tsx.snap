// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="flex-row justify-between items-center w-full p-5"
>
  <Text
    href="/"
    onPress={[Function]}
    role="link"
  >
    <View
      className="size-24 items-center justify-center rounded-full border-solid border-[#CA9D37] border"
    >
      <Image
        source={1}
        style={
          {
            "borderRadius": 9999,
            "height": "100%",
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
    </View>
  </Text>
  <View
    className="flex-row gap-2"
  >
    <Text
      href="/user/login"
      onPress={[Function]}
      role="link"
    >
      <View
        className="border-solid border-[#CA9D37] border bg-zinc-50 p-2 rounded-full"
      >
        <Text />
      </View>
    </Text>
    <Text
      href="/cart"
      onPress={[Function]}
      role="link"
    >
      <View
        className="border-solid border-[#CA9D37] border bg-zinc-50 p-2 rounded-full"
      >
        <Text />
      </View>
    </Text>
  </View>
</View>
`;
